org.jacoco.core.test/src/org/jacoco/core/test/InstrumentingLoader.java

```diff
@@ -34,6 +36,8 @@ public final class InstrumentingLoader extends ClassLoader {
 	private final RuntimeData data;
 	private final Instrumenter instrumenter;
 
+	private final ArrayList<String> instrumentedClasses;
```

Comment:
```suggestion
	private final Collection<String> instrumentedClasses;
```

Or maybe even `Set`? This probably the expected semantic.